### ISSUE TITLE
[Bug-ish] HintSignAsyncGenerator not really the same "sign" as a Generator

### ISSUE THREAD

**Issue by tlambert03** [beartype/beartype#138]
(whoops, sorry for empty first post)

In working on #136, I noticed that `HintSignAsyncGenerator` is classified as a `Generator` (which, of course, it is) ... however, it doesn't really behave the same as typing.Generator.  It expects [*two*](https://docs.python.org/3/library/typing.html#typing.AsyncGenerator) parameters instead of the normal three for a [generator](https://docs.python.org/3/library/typing.html#typing.Generator).  This appears to be the only sign in the list that has a mismatch:

<details>

```python
In [18]: from beartype._data.hint.pep.sign import datapepsigns as signs

In [19]: import typing

In [20]: for name, sign in vars(signs).items():
    ...:     if not name.startswith('HintSign'):
    ...:         continue
    ...:     try:
    ...:         t_obj = getattr(typing, name[8:])
    ...:     except AttributeError:
    ...:         continue
    ...:     if not hasattr(t_obj, '_nparams'):
    ...:         continue
    ...:     t_obj2 = getattr(typing, sign.name)
    ...:     print(name, sign.name, t_obj._nparams, t_obj2._nparams)
```

```
HintSignCallable Callable 2 2
HintSignTuple Tuple -1 -1
HintSignType Type 1 1
HintSignAbstractSet AbstractSet 1 1
HintSignByteString ByteString 0 0
HintSignContainer Container 1 1
HintSignContextManager ContextManager 1 1
HintSignHashable Hashable 0 0
HintSignItemsView ItemsView 2 2
HintSignIterable Iterable 1 1
HintSignIterator Iterator 1 1
HintSignKeysView KeysView 1 1
HintSignMapping Mapping 2 2
HintSignMappingView MappingView 1 1
HintSignMutableMapping MutableMapping 2 2
HintSignMutableSequence MutableSequence 1 1
HintSignMutableSet MutableSet 1 1
HintSignSequence Sequence 1 1
HintSignSized Sized 0 0
HintSignValuesView ValuesView 1 1
HintSignAwaitable Awaitable 1 1
HintSignAsyncIterator Iterator 1 1
HintSignAsyncIterable Iterable 1 1
HintSignCoroutine Coroutine 3 3
HintSignCollection Collection 1 1
HintSignAsyncGenerator Generator 2 3   # <------- Outlier!
HintSignAsyncContextManager ContextManager 1 1
HintSignReversible Reversible 1 1
HintSignChainMap ChainMap 2 2
HintSignCounter Counter 1 1
HintSignDeque Deque 1 1
HintSignDict Dict 2 2
HintSignDefaultDict DefaultDict 2 2
HintSignList List 1 1
HintSignOrderedDict OrderedDict 2 2
HintSignSet Set 1 1
HintSignFrozenSet FrozenSet 1 1
HintSignGenerator Generator 3 3
HintSignMatch Match 1 1
HintSignPattern Pattern 1 1
```

</details>

I know it's not in keeping with the rest of the patterns, but given the difference in usage parameters, would you consider a PR that changes:

```py
HintSignAsyncGenerator = _HintSign(name='Generator')
# to
HintSignAsyncGenerator = _HintSign(name='AsyncGenerator')
```

**Comment by leycec** [comment]
**Absolutely.** Thanks so much for yet another fast catch, @tlambert03. As you surmise, that's an appalling typo and definitely *not* the intended behaviour. Hallowed be thy username for finding this.

Thankfully, you already fixed this in #136. Phenomenal stuff! We'll close this out as soon as #136 lands. You rock the typing Casbah, bro. :man_singer:

**Comment by leycec** [comment]
Resolved by https://github.com/beartype/beartype/commit/7809ea2ae886dea168eac8cc636f690cfba6f1d7. Praise be to @tlambert03 for he is generous in his wisdom.
